spyder/widgets/sourcecode/api/decoration.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
```

Comment:
You have `sourcecode/api/decoration.py` and `sourcecode/api/decorations.py`, which is confusing.

Since this only contains a helper class, we could move it to `sourcecode/utils/decoration.py` or something like that.